realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -629,6 +655,25 @@ void checkIntValueIsLegal(long columnIndex, long rowToUpdate, long value) {
         }
     }
 
+    // check if it is ok to use null value for given row and column.
+    void checkNullValueIsLegal(long columnIndex, long rowToUpdate) {
```

Comment:
The name or implementation of this method is wrong since it only checks for primary key columns.
From the context it looks like the implementation i.e. it should also check for non-primary key columns.